packages/core/src/utils/withFaststore.js

```diff
@@ -0,0 +1,135 @@
+const path = require('path')
+const finalConfig = require('../../discovery.config')
+
+/**
+ * @type {AnotateWebpack}
+ * https://github.com/vercel/next.js/discussions/78170 - Do not use webpack aliases
+ */
+const withAliases = (aliases) => (userConfig) => (baseConfig, _context) => {
+  const config = { ...baseConfig, ...userConfig?.(baseConfig, _context) }
+
+  Object.entries(aliases).forEach(([k, v]) => {
+    config.resolve.alias[k] = v
+  })
+
+  return config
+}
+
+/**
+ * @type {AnotateWebpack}
+ */
+const withCamelCaseCss = (userConfig) => (baseConfig, _context) => {
+  const config = { ...baseConfig, ...userConfig?.(baseConfig, _context) }
```

Comment:
Are this objects always flat? Or do we expect them to be nested? If that's the case, deepmerge would be a great fit.